.circleci/real_config.yml

```diff
@@ -112,85 +112,74 @@ commands:
       - run: git submodule sync
       - run: git submodule update --init
 
-  skip-if-docs-only:
+  add-and-fetch-upstream:
     steps:
       - run: git remote add upstream https://github.com/determined-ai/determined
       - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for docs-only changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = "docs" ] ; then
-              echo "docs-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected non-docs-only changes: $DIFF_DIRS"
-            fi
+
+  skip-if-dir:
+    parameters:
+      dir:
+        type: string
+      not:
+        type: boolean
+        default: false
+    steps:
+      - unless:
+          condition: << parameters.dir >>
+          steps:
+            - run:
+                name: skip if << parameters.dir >>-only changes
+                command: |
+                  MERGE_BASE="$(git merge-base upstream/main HEAD)"
+                  DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
+                  if [ "$DIFF_DIRS" = "<< parameters.dir >>" ] ; then
+                    echo "<< parameters.dir >>-only change detected, halting job now"
+                    circleci-agent step halt
+                  else
+                    echo "detected non-<< parameters.dir >>-only changes: $DIFF_DIRS"
+                  fi
+      - when:
+          condition: << parameters.dir >>
+          steps:
+            - run:
+                name: skip if not << parameters.dir >> changes
+                command: |
+                  MERGE_BASE="$(git merge-base upstream/main HEAD)"
+                  DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
+                  if [[ "$DIFF_DIRS" == *"<< parameters.dir >>"* ]]; then
```

Comment:
Maybe not the right time to change this, but I've always disliked this.  DIFF_DIRS should probably be split up into an array, and then the array should be searched for the dir being compared.  Say there's `docs` and `postdocsamples` or `pseudocsi`.  Ok, bad example, but I still hate wildcards like that with no delimiters.  Maybe add a TODO or a warning or something in there?  It's probably unreasonable to actually split the string up and implement a shell-native linear search in this PR (or to generate an associative array, or...)

Reply:
okay good call! i'll add a comment but i'll try and avoid changing too much at once

Reply:
Yes please - small PRs are testable PRs. :)

Reply:
No need to mash everything together and then go back and search it -- just pipe `git diff-tree` to `grep`. Or use `git diff --quiet "$MERGE_BASE" HEAD -- "<< parameters.dir >>"`.

Reply:
Making git do the work instead of generating extra output just to immediately discard most of that output sounds like a huge improvement. Even though nobody knows how git actually works.